components/var_dumper/advanced.rst

```diff
@@ -240,3 +254,51 @@ properties not in the class declaration).
 .. tip::
 
     Before writing your own casters, you should check the existing ones.
+
+Adding semantics with metadata
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 3.2
+
+    As of Symfony 3.2, casters can attach metadata attributes to
+    :class:`Symfony\\Component\\VarDumper\\Cloner\\Stub` objects to inform
+    dumpers about the precise type of the dumped values.
+
+Since casters are hooked on specific classes or interfaces, they know about the
+objects they manipulate. By altering the ``$stub`` object (the third argument of
+any casters), one can transfer this knowledge to the resulting ``Data`` object,
+thus to dumpers. To help you do this (see the source code for how it works),
+the component comes with a set of wrappers for common additional semantics. You
+can use:
+
+ * :class:`Symfony\\Component\\VarDumper\\Caster\\ConstStub` to wrap a value that is
+   best represented by a PHP constant;
+ * :class:`Symfony\\Component\\VarDumper\\Caster\\ClassStub` to wrap a PHP identifier
+   (*i.e.* a class name, a method name, an interface, *etc.*);
+ * :class:`Symfony\\Component\\VarDumper\\Caster\\CutStub` to replace big noisy
+   objects/strings/*etc.* by ellipses;
+ * :class:`Symfony\\Component\\VarDumper\\Caster\\CutArrayStub` to keep only some
+   useful keys of an array;
+ * :class:`Symfony\\Component\\VarDumper\\Caster\\EnumStub` to wrap a set of virtual
+   values (*i.e.* values that do not exist as properties in the original PHP data
+   structure, but are worth listing alongside with real ones);
+ * :class:`Symfony\\Component\\VarDumper\\Caster\\LinkStub` to wrap strings that can
+   be turned into links by dumpers;
+ * :class:`Symfony\\Component\\VarDumper\\Caster\\TraceStub` and their
+ * :class:`Symfony\\Component\\VarDumper\\Caster\\FrameStub` and
+ * :class:`Symfony\\Component\\VarDumper\\Caster\\ArgsStub` relatives to wrap PHP
+   traces (used by :class:`Symfony\\Component\\VarDumper\\Caster\\ExceptionCaster`).
+
+For example, if you know that your ``Foo`` objects have a ``target`` property
```

Comment:
"your `Product` objects have a ``brochure`` property that holds a file name or URL, [...]"

And then also modify the example. Things are always a lot less complex when using real world examples :)